content/blog/2020-09-28-september-20-community-gems.md

```diff
@@ -0,0 +1,198 @@
+so DVC can version it without resorting to time-consuming data transfer
+operations.
+
+External outputs are configured using using `dvc add --external`
+([there are plenty more details and tips about managing external data in our docs](https://dvc.org/doc/user-guide/managing-external-data)).
+
+One useful note: when you use an external output, you'll want to set up an
+[external cache](https://dvc.org/doc/use-cases/shared-development-server#configure-the-external-shared-cache)
+in the same remote location. This is because the default cache location (in your
+local workspace) no longer makes sense when the dataset never "visits" your
+local workspace! An external cache works largely the same as a typical cache in
+your workspace- DVC will try to use the cache/link type you set up in the remote
+location where the external data and cache are.
+
+Now, let's talk about **external dependencies**. An external dependency might
+come into play when you run a DVC pipeline stage that depends on a file or
+directory in a remote location. For example, a pipeline stage could depend on a
+`model.pkl` file in your S3 bucket:
+
+```dvc
+$ dvc run -d s3://mybucket/model.pkl
+```
+
+DVC connects to the remote location (here, your S3 bucket) to check whether or
+not the indicated file (`model.pkl`) has changed since the last time `dvc repro`
+was run, but that's it. There's no caching involved.
+
+## CML questions
+
+### [Q: How can I use CML with my own Docker container?](https://discordapp.com/channels/485586884165107732/728693131557732403/757553135840526376)
+
+In many of our CML docs and videos, we've shown how to get CML on your CI
+(continuous integration) runner via a Docker container that comes with
+everything installed. But this is not the only way to use CML, especially if you
+want workflows to run in your own Docker container.
+
+You can install CML via `npm`, either in your own Docker container or in your CI
+workflow (i.e., in your GitHub Actions `.yaml` or GitLab CI `.yml` workflow
+file).
+
+To install CML as a package, you'll want to run:
+
+```bash
+$ npm i -g @dvcorg/cml
+```
+
+Note that you may need to install additional dependencies if you want to use DVC
+plots and Vega-Lite commands:
+
+```bash
+$ sudo apt-get install -y libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev \
+          librsvg2-dev libfontconfig-dev
+$ npm install -g vega-cli vega-lite
+```
+
+If you're installing CML as part of your workflow, you may need to install Node
+first-
+[check out our docs](https://github.com/iterative/cml#install-cml-as-a-package)
+for how to do this in GitHub Actions and GitLab CI.
+
+### [Q: After running a GitHub Action workflow that runs a DVC pipeline, I want to save the output of the pipeline. Why doesn't CML automatically save the output?](https://discordapp.com/channels/485586884165107732/728693131557732403/757686601953312988)
+
+What you're talking about sounds like an auto-commit: a `git commit` at the end
```

Comment:
should add some context here- why commit is needed at all?

Reply:
Added 

```
By design, artifacts generated in a CI workflow aren't saved anywhere- they disappear as soon as the runner shuts down. So a DVC pipeline executed in your CI system might produce outputs, like transformed datasets and model files, that will be lost at the end of the run. If you want to save them, there are a few methods. 
```